From 31834aaa4e2a26d8d1f6b36703bb35cfdb8fc98c Mon Sep 17 00:00:00 2001
Message-Id: <31834aaa4e2a26d8d1f6b36703bb35cfdb8fc98c.1651196605.git.geliang.tang@suse.com>
From: Yang Yingliang <yangyingliang@huawei.com>
Date: Thu, 6 Jan 2022 15:54:48 +0800
Subject: [PATCH] ACPI: pfr_update: Fix return value check in pfru_write()
Git-commit: 31834aaa4e2a26d8d1f6b36703bb35cfdb8fc98c
Patch-mainline: v5.17-rc1
References: bsn#210
Modified-by-SEL: No

In case of error, memremap() returns NULL pointer not
ERR_PTR(). The IS_ERR() test in the return value check
should be replaced with NULL test.

Fixes: 0db89fa243e5 ("ACPI: Introduce Platform Firmware Runtime Update device driver")
Reported-by: Hulk Robot <hulkci@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Acked-by: Chen Yu <yu.c.chen@intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 drivers/acpi/pfr_update.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/acpi/pfr_update.c b/drivers/acpi/pfr_update.c
index 149b5b2530b9..6bb0b778b5da 100644
--- a/drivers/acpi/pfr_update.c
+++ b/drivers/acpi/pfr_update.c
@@ -460,8 +460,8 @@ static ssize_t pfru_write(struct file *file, const char __user *buf,
 	/* map the communication buffer */
 	phy_addr = (phys_addr_t)((buf_info.addr_hi << 32) | buf_info.addr_lo);
 	buf_ptr = memremap(phy_addr, buf_info.buf_size, MEMREMAP_WB);
-	if (IS_ERR(buf_ptr))
-		return PTR_ERR(buf_ptr);
+	if (!buf_ptr)
+		return -ENOMEM;
 
 	if (!copy_from_iter_full(buf_ptr, len, &iter)) {
 		ret = -EINVAL;
-- 
2.34.1

